### PR TITLE
Fix handling empty values of NO_COLOR and FORCE_COLOR

### PR DESCRIPTION
**Pull request by mgorny**
Fix handling NO_COLOR and FORCE_COLOR environment variables to correctly be ignored when they are set to an empty value, as defined in the specification:

> Command-line software which adds ANSI color to its output by default
> should check for a NO_COLOR environment variable that, when present
> *and not an empty string* (regardless of its value), prevents
> the addition of ANSI color.

(emphasis mine, https://no-color.org/)

The same is true of FORCE_COLOR, https://force-color.org/.

----

- [ ] Include documentation when adding new features.
- [x] Include new tests or update existing tests when applicable.
- [X] Allow maintainers to push and squash when merging my commits. Please uncheck this if you prefer to squash the commits yourself.

If this change fixes an issue, please:

- [ ] Add text like ``closes #XYZW`` to the PR description and/or commits (where ``XYZW`` is the issue number). See the [github docs](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more information.

Unless your change is trivial or a small documentation fix (e.g., a typo or reword of a small section) please:

- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/main/changelog/README.rst) for details.

  Write sentences in the **past or present tense**, examples:

  * *Improved verbose diff output with sequences.*
  * *Terminal summary statistics now use multiple colors.*

  Also make sure to end the sentence with a `.`.

- [x] Add yourself to `AUTHORS` in alphabetical order.

### CODE REVIEW

File: testing/io/test_terminalwriter.py [hunk 1]
```diff
@@ -213,6 +213,12 @@ def test_should_not_do_markup_NO_COLOR_and_FORCE_COLOR(
     assert_color_not_set()
 
 
+def test_empty_NO_COLOR_ignored(monkeypatch: MonkeyPatch) -> None:
+    monkeypatch.setitem(os.environ, "NO_COLOR", "")
+    monkeypatch.setitem(os.environ, "FORCE_COLOR", "1")
```
Comment (RonnyPfannschmidt) [inline testing/io/test_terminalwriter.py:218]: When setting both,the test doesn't actually cover all cases plus off hand I don't recall if our checks consider externally set variables

**Review by RonnyPfannschmidt — COMMENTED** [review]
Good catch, thanks for ensuring we match the spec

Please ensure the tests actually cover the cases

File: testing/io/test_terminalwriter.py (hunk 1 above)
Comment (mgorny) [reply testing/io/test_terminalwriter.py]: I'm sorry but I don't understand the second part of your sentence. Could you rephrase, please?

**Comment by mgorny** [comment]
I've done some refactoring and covered all cases of `FORCE_COLOR` and `NO_COLOR` (unset, set to empty, set to `1`).

**Review by bluetech — APPROVED** [review]
Thanks!

**Comment by mgorny** [comment]
Thanks!
